core/templates/components/main-nav/main-nav.twig

```diff
@@ -0,0 +1,44 @@
+{%- if markup is not empty %}
+  {# If custom markup is provided, emit it as is #}
+  {{ markup }}
+{% else -%}
+  {# If no custom markup, generate markup based on structured data #}
+<nav {{- attributes }} class="su-main-nav {{ modifier_class }}" aria-label="Main">
+  <button class="su-main-nav__toggle" aria-expanded="false">{{ toggle_text }}</button>
+  {% if lv1 is not empty -%}
+    <ul class="su-main-nav__menu-lv1">
+      {% for item in lv1 %}
+        {% set is_parent = item.lv2 is empty ? false : true %}
+
+        <li class="{% if is_parent %}su-main-nav__item--parent{% endif %} {% if item.current %}su-main-nav__item--current{% endif %}">{% spaceless %}
+          <a href="{{ item.href }}" {% if item.target is not empty %}target="{{ item.target }}"{% endif %}>{{ item.text }}</a>{% endspaceless %}
+
+          {% if is_parent -%}
+            <ul class="su-main-nav__menu-lv2">
+              {%- for item_lv2 in item.lv2 -%}
+                {% spaceless %}<li class="{% if item_lv2.current %}su-main-nav__item--current{% endif %}">
+                  <a href="{{ item_lv2.href }}" {% if item_lv2.target is not empty %}target="{{ item_lv2.target }}"{% endif %}>{{ item_lv2.text }}</a>
+                </li>{% endspaceless %}
+
+              {% endfor %}
+            </ul>
+          {% endif -%}
+        </li>
+      {% endfor %}
+    </ul>
+  {% endif %}
+  {% if "su-main-nav--mobile-search" in modifier_class %}
+    {# no blessed idea why {% include ... with mobile_search %} doesn't work, but it doesn't, so reference each variable individually :( #}
```

Comment:
This makes sense as you have to explicitly pass through the value you want to the site-search.twig file. You cannot pass through just the `mobile_search` object as it won't define the appropriate variables in the included template. 

One addition you may want to include here is to use `with only` as we may get unwanted variable leakage in to the included template.